### ISSUE TITLE
The "Squiz.WhiteSpace.OperatorSpacing" and new lines issue

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#348]
When code around mathematical operation or any other operator is split across several lines then new line symbol or any other whitespace (except comment) should be considered as valid whitespace.

```php
$original_amount = $order->GetDBField('SubTotal') + $order->GetDBField('ShippingCost') +
			$order->GetDBField('VAT') + $order->GetDBField('ProcessingFee') + $order->GetDBField('InsuranceFee') -
			$order->GetDBField('GiftCertificateDiscount');
```

Currently one space specifically is asserted to be present.

**Comment by euoia** [comment]
I added a PR which adds an option for this. Please let me know if I need to add or modify any tests.

**Comment by gsherwood** [comment]
I've added a sniff option `ignoreNewlines` that will not check the spacing before or after an operator if a newline is detected instead. Note that this sniff will not enforce specific indentation or that code once it is on a newline, and will never recommend using a newline as that is not its goal.

To ignore newlines, put the following into a ruleset.xml file:
```xml
<rule ref="Squiz.WhiteSpace.OperatorSpacing">
  <properties>
    <property name="ignoreNewlines" value="true"/>
  </properties>
</rule>
```

**Comment by JeroenDeDauw** [comment]
\o/ Thanks @gsherwood!

**Comment by euoia** [comment]
That's great - thanks @gsherwood. Any chance you could tell me when we're likely to see the next release of phpcs that includes this change?

**Comment by aik099** [comment]
Maybe there is a roadmap somewhere? I see a lot of features being added lately. Maybe release once in 2 months would be great.

**Comment by JeroenDeDauw** [comment]
Looking at the [previous releases](https://github.com/squizlabs/PHP_CodeSniffer/releases), it seems like they are made at sane intervals.

**Comment by gsherwood** [comment]
I do a release when I think the code is stable enough. Dumping a bunch of bug fixes and new features into the master branch and then just tagging it isn't something I like to do. PHPCS gets a heavy workout internally first to try and catch the majority of problems, which has been very helpful this time in catching a few infinite loops, so I think that process is working.

The current code is very near a release (probably next week at this stage). I was trying to get some more fixes in, but they are looking like they will be bigger than expected, so I'll have to leave them for now.

**Comment by euoia** [comment]
That's really helpful to know, thanks.
